lib/TH/THFilePrivate.h

```diff
@@ -32,6 +40,7 @@ struct THFileVTable
     size_t (*writeLong)(THFile *self, long *data, size_t n);
     size_t (*writeFloat)(THFile *self, float *data, size_t n);
     size_t (*writeDouble)(THFile *self, double *data, size_t n);
+    size_t (*writeHalf)(THFile *self, half *data, size_t n);
```

Comment:
re .h files: considering this flag temporary, I left the places that won't actually break things, unguarded. I can guard them, too, if you think having compile-time error here is better than 'accidentally' working code.